### ISSUE TITLE
quant not exiting with correct error code

### ISSUE THREAD

**Issue by ATpoint** [COMBINE-lab/alevin-fry#95]
Hi Rob,

I think [here](https://github.com/COMBINE-lab/alevin-fry/blob/master/src/main.rs#L553-L554) there is an `std::process::exit(1);` missing. I just had a run with this error but it still exited with error code 0 rather than 1. Version 0.8.0 

```
alevin-fry quant -t 1 -i sample1 -o quant_fry --tg-map splici_t2g_3col.tsv --resolution cr-like --use-mtx && echo "OK" || echo "FAIL"
```

...returned this error (because I messed up providing the permit files properly) but also the "OK" echo. Can you double-check, I think this should be an exit of 1.

**Comment by rob-p** [comment]
Hi @ATpoint,

  You're right that the exit code isn't set properly in this case.  At some point, we switched from the "historic" unit return type for main (where all non-zero exit codes had to result from a `std::process::exit`) to the more modern `Result<()>` return type for `main`. Some code paths (like this one, I believe) slipped through the cracks. It seems it wants to print the error message, but that doesn't prevent it from returning 0 (i.e. returning `Ok(())` from main).  I think the right thing to do here is to ensure in this path that we return `Err(<appropriate error message here>)`, which will both show the message and force the exit code to 1.

Thanks!
Rob

**Comment by rob-p** [comment]
Fixed in 0.8.1.
